HPCC-24120 Azure storage examples

jakesmith opened the pull request:
HPCC-24120 Azure storage examples

+ values-auto-azurefile.yaml

Simple overrides of hpcc storageClass' to use "azurefile".
PVC's will be auto provisioned with default reclaimPolicy=Delete

+ hpcc-azurefile helm charts

Helm charts that defined StorageClass' for each HPCC storage type,
Persistent Volume Claim's for each to use these StorageClass'

+ values-retained-azurefile.yaml

Overrides to use the PVC's provided by the "hpcc-azurefile" helm
charts.

Signed-off-by: Jake Smith <jake.smith@lexisnexisrisk.com>

<!-- Thank you for submitting a pull request to the HPCC project

 PLEASE READ the following before proceeding.

 This project only accepts pull requests related to open JIRA issues.
 If suggesting a new feature or change, please discuss it in a JIRA issue first.
 If fixing a bug, there should be an issue describing it with steps to reproduce.
 The title line of the pull request (and of each commit within it) should refer to the
 associated issue using the format:

 HPCC-nnnnn Short description of issue

 This will allow the Jira ticket to be automatically updated to refer to this pull request,
 and will ensure that the automatically-generated changelog is properly formatted.
 Where a pull request contains a single commit the pull request title will be set automatically,
 assuming that the commit has followed the proper guidelines.

 Please go over all the following points, and put an `x` in all the boxes that apply. You may find
 it easier to press the 'Create' button first then click on the checkboxes to edit the comment.
-->

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

## Testing:
<!-- Please describe how this change has been tested.-->

<!-- Thank you for taking the time to submit this pull request and to answer all of the above-->

jakesmith commented:
@richardkchapman @ghalliday - this is the setup I use for azure storage

HPCCSmoketest commented:
Process of PR-13770, label: azure-persistent-storage starts now.
The reason of this test is: Did not build yet.
Commit ID: f5c1c7773b0771b5caf8ff5a09a2136c001a3358
Estimated completion time is ~0.9 hour(s)
centos 7.8.2003 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-116.ca-central-1.compute.internal

jakesmith commented:
NB: this approach means this separate helm chart must be installed first, to provision the PVC's.
e.g.:
```
helm install azure-storage ./examples/azure-storage/
```

Such that you end up having, e.g.:
```
$ kubectl get pvc
NAME            STATUS   VOLUME                                     CAPACITY   ACCESS MODES   STORAGECLASS   AGE
hpcc-dali-pvc   Bound    pvc-753eedf1-32b9-4cff-b061-9a5352553c0a   1Gi        RWO            hpcc-dali-sc   48m
hpcc-data-pvc   Bound    pvc-1cd63767-3d70-4497-a00b-ab5394d052ee   1Gi        RWX            hpcc-data-sc   48m
hpcc-dll-pvc    Bound    pvc-67482915-ea42-41db-8b5c-10d7ebfd6bb1   1Gi        RWX            hpcc-dll-sc    48m
```
pv's:
```
$ kubectl get pv | grep jakes
pvc-1cd63767-3d70-4497-a00b-ab5394d052ee   1Gi        RWX            Retain           Bound    jakes-namespace/hpcc-data-pvc                             hpcc-data-sc            50m
pvc-67482915-ea42-41db-8b5c-10d7ebfd6bb1   1Gi        RWX            Retain           Bound    jakes-namespace/hpcc-dll-pvc                              hpcc-dll-sc             50m
pvc-753eedf1-32b9-4cff-b061-9a5352553c0a   1Gi        RWO            Retain           Bound    jakes-namespace/hpcc-dali-pvc                             hpcc-dali-sc            50m
```

and then once you also helm install hpcc:

```
$ helm list
NAME                    NAMESPACE       REVISION        UPDATED                                 STATUS          CHART                           APP VERSION
hpcc-azure-storage      jakes-namespace 1               2020-05-21 19:10:23.037549466 +0100 BST deployed        hpcc-azure-storage-0.1.0        0.1.0
mycluster               jakes-namespace 1               2020-05-21 19:13:53.411532766 +0100 BST deployed        hpcc-0.1.0                      0.1.0
```

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
OS: centos 7.8.2003 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-116.ca-central-1.compute.internal
Sha: f5c1c7773b0771b5caf8ff5a09a2136c001a3358
Build: success
Milestone:Install hpccsystems-platform-community_7.8.17-closedown0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | elaps | 
|---|---|---|---|---|---|---|
|unittest |120 |120 |0 |0 |0 |53 sec |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |1 sec |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |9 sec |



Regression test result:

| phase | total | pass | fail | elaps | 
|---|---|---|---|---|
|setup (hthor) |11 |11 |0 |30 sec  (00:00:30) |
|setup (thor) |11 |11 |0 |44 sec  (00:00:44) |
|setup (roxie) |11 |11 |0 |19 sec  (00:00:19) |
|test (hthor) |874 |874 |0 |570 sec  (00:09:30) |
|test (thor) |812 |812 |0 |998 sec  (00:16:38) |
|test (roxie) |953 |953 |0 |715 sec  (00:11:55) |


HPCC Stop: OK
HPCC Uninstall: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|    9 sec (00:00:09) |  925 sec (00:15:25) |  122 sec (00:02:02) |   22 sec (00:00:22) |   17 sec (00:00:17) | 2637 sec (00:43:57) |   18 sec (00:00:18) | 3750 sec (01:02:30) |

jakesmith committed:
HPCC-24120 Azure storage examples

+ values-auto-azurefile.yaml

Simple overrides of hpcc storageClass' to use "azurefile".
PVC's will be auto provisioned with default reclaimPolicy=Delete

+ hpcc-azurefile helm charts

Helm charts that defined StorageClass' for each HPCC storage type,
Persistent Volume Claim's for each to use these StorageClass'

+ values-retained-azurefile.yaml

Overrides to use the PVC's provided by the "hpcc-azurefile" helm
charts.

Signed-off-by: Jake Smith <jake.smith@lexisnexisrisk.com>

HPCCSmoketest commented:
Process of PR-13770, label: azure-persistent-storage starts now.
The reason of this test is: New commit.
Commit ID: 133b224edf0681ca2d0fb3034a53724da68dd0a7
Estimated completion time is ~0.9 hour(s)
centos 7.8.2003 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-116.ca-central-1.compute.internal

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
OS: centos 7.8.2003 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-116.ca-central-1.compute.internal
Sha: 133b224edf0681ca2d0fb3034a53724da68dd0a7
Build: success
Milestone:Install hpccsystems-platform-community_7.8.17-closedown0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | elaps | 
|---|---|---|---|---|---|---|
|unittest |120 |120 |0 |0 |0 |54 sec |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |2 sec |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |9 sec |



Regression test result:

| phase | total | pass | fail | elaps | 
|---|---|---|---|---|
|setup (hthor) |11 |11 |0 |29 sec  (00:00:29) |
|setup (thor) |11 |11 |0 |44 sec  (00:00:44) |
|setup (roxie) |11 |11 |0 |22 sec  (00:00:22) |
|test (hthor) |874 |874 |0 |567 sec  (00:09:27) |
|test (thor) |812 |812 |0 |1024 sec  (00:17:04) |
|test (roxie) |953 |953 |0 |716 sec  (00:11:56) |


HPCC Stop: OK
HPCC Uninstall: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|    8 sec (00:00:08) |  913 sec (00:15:13) |  123 sec (00:02:03) |   24 sec (00:00:24) |   17 sec (00:00:17) | 2642 sec (00:44:02) |   18 sec (00:00:18) | 3745 sec (01:02:25) |

HPCCSmoketest commented:
Process of PR-13770, label: azure-persistent-storage starts now.
The reason of this test is: New commit.
Commit ID: 85c757a291005798f841eedcf77821796f062743
Estimated completion time is ~0.9 hour(s)
centos 7.8.2003 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-116.ca-central-1.compute.internal

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
OS: centos 7.8.2003 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-116.ca-central-1.compute.internal
Sha: 85c757a291005798f841eedcf77821796f062743
Build: success
Milestone:Install hpccsystems-platform-community_7.8.17-closedown0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | elaps | 
|---|---|---|---|---|---|---|
|unittest |120 |120 |0 |0 |0 |53 sec |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |2 sec |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |8 sec |



Regression test result:

| phase | total | pass | fail | elaps | 
|---|---|---|---|---|
|setup (hthor) |11 |11 |0 |30 sec  (00:00:30) |
|setup (thor) |11 |11 |0 |44 sec  (00:00:44) |
|setup (roxie) |11 |11 |0 |20 sec  (00:00:20) |
|test (hthor) |874 |874 |0 |570 sec  (00:09:30) |
|test (thor) |812 |812 |0 |1009 sec  (00:16:49) |
|test (roxie) |953 |953 |0 |705 sec  (00:11:45) |


HPCC Stop: OK
HPCC Uninstall: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|    9 sec (00:00:09) |  934 sec (00:15:34) |  123 sec (00:02:03) |   23 sec (00:00:23) |   17 sec (00:00:17) | 2617 sec (00:43:37) |   17 sec (00:00:17) | 3740 sec (01:02:20) |

ghalliday reviewed (commented):
@jakesmith a few fairly trivial comments

jakesmith commented on dockerfiles/examples/azure/retained-storage/templates/dllserver-sc.yaml:
```diff
@@ -0,0 +1,9 @@
+apiVersion: storage.k8s.io/v1
```
yep, will rename

jakesmith commented on dockerfiles/examples/azure/retained-storage/values.yaml:
```diff
@@ -0,0 +1,10 @@
+
+dllStorage:
+  size: 256Mi
+
```
agree, will add.

jakesmith commented on dockerfiles/examples/azure/retained-storage/values.yaml:
```diff
@@ -0,0 +1,10 @@
+dataStorage:
+  size: 1Gi
+
+daliStorage:
```
I did, it got lost from this PR somehow. Will re-add.

jakesmith commented on dockerfiles/examples/azure/values-storage-retained-azurefile.yaml:
```diff
@@ -0,0 +1,16 @@
+# NB: The "retained-storage" helm charts should be installed 1st.
+# The "retrained-storage" helm chart provisions and creates the
+# PVC's required by HPCC.
+# There values are dynamic based on the install name.
```
agree, will change.

jakesmith commented on dockerfiles/examples/azure/values-storage-retained-azurefile.yaml:
```diff
@@ -0,0 +1,16 @@
+# Overrides for azure storage in hpcc, using custom StorageClass'
```
I don't know.. when pluralizing a reserved word like this, I usually try to leave it intact, but no idea what is convention or correct in these circumstances.

ghalliday replied:
' normally implies possession.  @richardkchapman votes?

jakesmith commented on dockerfiles/examples/azure/values-storage-retained-azurefile.yaml:
```diff
@@ -0,0 +1,16 @@
+# Overrides for azure storage in hpcc, using custom StorageClass'
+# NB: The "retained-storage" helm charts should be installed 1st.
+# The "retrained-storage" helm chart provisions and creates the
```
will correct

jakesmith commented on dockerfiles/examples/azure/retained-storage/templates/_helpers.tpl:
```diff
@@ -0,0 +1,63 @@
+{{- end -}}
+{{- end -}}
+
+{{/*
```
I don't know, wasn't sure about other parts created by helm create either.
e.g. .helmignore
I think prob. ok to leave them in as standard.

jakesmith commented on dockerfiles/examples/azure/retained-storage/.helmignore:
```diff
@@ -0,0 +1,23 @@
+# Patterns to ignore when building packages.
+# This supports shell glob matching, relative path matching, and
```
maybe, I was trying to distinguish between it and the storage presented via 'values-storage-auto-azurefile.yaml' settings file.
Which simply overrides the storageClass in hpcc to use azurefile, and as a result auto provisions PV's that have a retainPolicy of "Delete".

But the helm chart could be called "storage" and it's meaning should probably be in an accompanying README

jakesmith commented:
@ghalliday - addressed comments above.

One question I had, was what do you think about the approach of generating the values file you 
need to plugin to hpcc when you install this? (i.e. via NOTES.txt)

Because the names are not static, but based on Release.Name (well fullname from _helpers.yaml), the override file that's required to be supplied to the HPCC helm chart is not static.

HPCCSmoketest commented:
Process of PR-13770, label: azure-persistent-storage starts now.
The reason of this test is: New commit.
Commit ID: cf1dfcc1673a3666687b984eacd5d260e0edf6e2
Estimated completion time is ~0.9 hour(s)
centos 7.8.2003 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-116.ca-central-1.compute.internal

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
OS: centos 7.8.2003 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-116.ca-central-1.compute.internal
Sha: cf1dfcc1673a3666687b984eacd5d260e0edf6e2
Build: success
Milestone:Install hpccsystems-platform-community_7.8.17-closedown0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | elaps | 
|---|---|---|---|---|---|---|
|unittest |120 |120 |0 |0 |0 |53 sec |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |2 sec |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |8 sec |



Regression test result:

| phase | total | pass | fail | elaps | 
|---|---|---|---|---|
|setup (hthor) |11 |11 |0 |28 sec  (00:00:28) |
|setup (thor) |11 |11 |0 |45 sec  (00:00:45) |
|setup (roxie) |11 |11 |0 |19 sec  (00:00:19) |
|test (hthor) |874 |874 |0 |574 sec  (00:09:34) |
|test (thor) |812 |812 |0 |1020 sec  (00:17:00) |
|test (roxie) |953 |953 |0 |703 sec  (00:11:43) |


HPCC Stop: OK
HPCC Uninstall: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|    8 sec (00:00:08) |  931 sec (00:15:31) |  123 sec (00:02:03) |   23 sec (00:00:23) |   18 sec (00:00:18) | 2626 sec (00:43:46) |   17 sec (00:00:17) | 3746 sec (01:02:26) |

HPCCSmoketest commented:
Process of PR-13770, label: azure-persistent-storage starts now.
The reason of this test is: New commit.
Commit ID: dfbd010bfd4df1bcb9cdb6c79a8cbdbd6e61181d
Estimated completion time is ~0.9 hour(s)
centos 7.8.2003 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-116.ca-central-1.compute.internal

jakesmith commented:
@ghalliday - rebased

HPCCSmoketest commented:
Process of PR-13770, label: azure-persistent-storage starts now.
The reason of this test is: Forced to re-test.
Commit ID: 4914d1bb6387d9744139db92ad062d61c75e9bbd
Estimated completion time is ~0.8 hour(s)
centos 7.6.1810 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-40.ca-central-1.compute.internal

ghalliday commented on dockerfiles/examples/azure/storage/values.schema.json:
```diff
@@ -0,0 +1,32 @@
+      "required": [ "size" ],
+      "properties": {
+        "size": {
+          "type": "string"
```
Is string the correct type?

jakesmith replied:
I think has to be because needs to take values like "3M", "3Mi", "3Gi"..

ghalliday replied:
Should possibly be something like:
    "io.k8s.apimachinery.pkg.api.resource.Quantity": {
      "oneOf": [
        {
          "type": "string"
        },
        {
          "type": "integer"
        }
      ]
    },

jakesmith replied:
unless there's a way to actually type check the format properly, I think string should suffice

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
OS: centos 7.8.2003 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-116.ca-central-1.compute.internal
Sha: dfbd010bfd4df1bcb9cdb6c79a8cbdbd6e61181d
Build: success
Milestone:Install hpccsystems-platform-community_7.8.19-closedown0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | elaps | 
|---|---|---|---|---|---|---|
|unittest |120 |120 |0 |0 |0 |54 sec |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |2 sec |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |8 sec |



Regression test result:

| phase | total | pass | fail | elaps | 
|---|---|---|---|---|
|setup (hthor) |11 |11 |0 |28 sec  (00:00:28) |
|setup (thor) |11 |11 |0 |45 sec  (00:00:45) |
|setup (roxie) |11 |11 |0 |20 sec  (00:00:20) |
|test (hthor) |874 |874 |0 |576 sec  (00:09:36) |
|test (thor) |812 |812 |0 |1024 sec  (00:17:04) |
|test (roxie) |953 |953 |0 |703 sec  (00:11:43) |


HPCC Stop: OK
HPCC Uninstall: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|    8 sec (00:00:08) |  904 sec (00:15:04) |  123 sec (00:02:03) |   22 sec (00:00:22) |   17 sec (00:00:17) | 2635 sec (00:43:55) |   18 sec (00:00:18) | 3727 sec (01:02:07) |

HPCCSmoketest commented:
Process of PR-13770, label: azure-persistent-storage starts now.
The reason of this test is: New commit.
Commit ID: 4914d1bb6387d9744139db92ad062d61c75e9bbd
Estimated completion time is ~0.9 hour(s)
centos 7.8.2003 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-116.ca-central-1.compute.internal

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
OS: centos 7.6.1810 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-40.ca-central-1.compute.internal
Sha: 4914d1bb6387d9744139db92ad062d61c75e9bbd
Build: success
Milestone:Install hpccsystems-platform-community_7.8.19-closedown0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | elaps | 
|---|---|---|---|---|---|---|
|unittest |120 |120 |0 |0 |0 |53 sec |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |1 sec |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |8 sec |



Regression test result:

| phase | total | pass | fail | elaps | 
|---|---|---|---|---|
|setup (hthor) |11 |11 |0 |28 sec  (00:00:28) |
|setup (thor) |11 |11 |0 |44 sec  (00:00:44) |
|setup (roxie) |11 |11 |0 |23 sec  (00:00:23) |
|test (hthor) |874 |874 |0 |311 sec  (00:05:11) |
|test (thor) |812 |812 |0 |761 sec  (00:12:41) |
|test (roxie) |953 |953 |0 |377 sec  (00:06:17) |


HPCC Stop: OK
HPCC Uninstall: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|    9 sec (00:00:09) |  651 sec (00:10:51) |  118 sec (00:01:58) |   22 sec (00:00:22) |   18 sec (00:00:18) | 1781 sec (00:29:41) |   18 sec (00:00:18) | 2617 sec (00:43:37) |

ghalliday commented on dockerfiles/examples/azure/values-storage-auto-azurefile.yaml:
```diff
@@ -0,0 +1,16 @@
+# Overrides for azure storage in hpcc.
```
I wonder if these should go in a separate subdirectory azure/config/... to group them together.

jakesmith replied:
perhaps, but then is might be less clear that subdirs are helm charts..
should the helm subchars go in a 'helm' subdir?

And I'm not sure about "values-storage-retained-azurefile.yaml" since the helm chart generates the correct Release named based values content in it's output...

jakesmith replied:
I propose putting helm charts in a "helm" subdir. and the value config examples under "config".
That will allow other examples to have multiple helm varieties if need be, and distinguishes between the subdirs that are helm charts vs the config subdir.

So for Azure we'd have:
```
examples/
-azure/
--helm/
---storage/
--config/
```

@ghalliday @richardkchapman thoughts?

ghalliday commented:
I think it looks good, a couple of comments.  Picky because this is going to be a template for various other examples.  The structure feels broadly correct.
I'm going to create a similar one for host mounted storage, which I think will fit in neatly.

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
OS: centos 7.8.2003 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-116.ca-central-1.compute.internal
Sha: 4914d1bb6387d9744139db92ad062d61c75e9bbd
Build: success
Milestone:Install hpccsystems-platform-community_7.8.19-closedown0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | elaps | 
|---|---|---|---|---|---|---|
|unittest |120 |120 |0 |0 |0 |54 sec |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |1 sec |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |9 sec |



Regression test result:

| phase | total | pass | fail | elaps | 
|---|---|---|---|---|
|setup (hthor) |11 |11 |0 |28 sec  (00:00:28) |
|setup (thor) |11 |11 |0 |44 sec  (00:00:44) |
|setup (roxie) |11 |11 |0 |20 sec  (00:00:20) |
|test (hthor) |874 |874 |0 |567 sec  (00:09:27) |
|test (thor) |812 |812 |0 |1007 sec  (00:16:47) |
|test (roxie) |953 |953 |0 |709 sec  (00:11:49) |


HPCC Stop: OK
HPCC Uninstall: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|    9 sec (00:00:09) |  925 sec (00:15:25) |  123 sec (00:02:03) |   22 sec (00:00:22) |   17 sec (00:00:17) | 2638 sec (00:43:58) |   18 sec (00:00:18) | 3752 sec (01:02:32) |

richardkchapman commented on dockerfiles/examples/azure/storage/Chart.yaml:
```diff
@@ -0,0 +1,21 @@
+apiVersion: v2
+name: storage
+description: A Helm chart for Kubernetes
```
A better description would be good

HPCCSmoketest commented:
Process of PR-13770, label: azure-persistent-storage starts now.
The reason of this test is: New commit.
Commit ID: 33575b5482cae5b11f303f4bc6c83107880f4577
Estimated completion time is ~0.9 hour(s)
centos 7.8.2003 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-116.ca-central-1.compute.internal

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
OS: centos 7.8.2003 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-116.ca-central-1.compute.internal
Sha: 33575b5482cae5b11f303f4bc6c83107880f4577
Build: success
Milestone:Install hpccsystems-platform-community_7.8.19-closedown0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | elaps | 
|---|---|---|---|---|---|---|
|unittest |120 |120 |0 |0 |0 |52 sec |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |1 sec |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |9 sec |



Regression test result:

| phase | total | pass | fail | elaps | 
|---|---|---|---|---|
|setup (hthor) |11 |11 |0 |29 sec  (00:00:29) |
|setup (thor) |11 |11 |0 |45 sec  (00:00:45) |
|setup (roxie) |11 |11 |0 |20 sec  (00:00:20) |
|test (hthor) |874 |874 |0 |573 sec  (00:09:33) |
|test (thor) |812 |812 |0 |1012 sec  (00:16:52) |
|test (roxie) |953 |953 |0 |710 sec  (00:11:50) |


HPCC Stop: OK
HPCC Uninstall: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|    8 sec (00:00:08) |  930 sec (00:15:30) |  124 sec (00:02:04) |   23 sec (00:00:23) |   18 sec (00:00:18) | 2626 sec (00:43:46) |   19 sec (00:00:19) | 3748 sec (01:02:28) |
